pandas/core/strings.py

```diff
@@ -564,7 +564,7 @@ def str_replace(arr, pat, repl, n=-1, case=None, flags=0, regex=True):
             # add case flag, if provided
             if case is False:
                 flags |= re.IGNORECASE
-        if is_compiled_re or len(pat) > 1 or flags or callable(repl):
+        if is_compiled_re or len(pat) > 0 or flags or callable(repl):
```

Comment:
Do we even need the `len(pat)` condition? Can it just be `pat` instead?

Reply:
It can be just `pat`, the only issue is case `pat` is empty.

Reply:
Wouldn’t that be False in either case? If so shouldn’t need the len expression